awscli/customizations/autoprompt/history.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import json
+import logging
+import os
+
+from prompt_toolkit.completion import Completion, Completer
+from prompt_toolkit.history import FileHistory
+
+from awscli.autocomplete.completer import CompletionResult
+from awscli.autocomplete.filters import fuzzy_filter
+
+
+LOG = logging.getLogger(__name__)
+
+
+class HistoryDriver(FileHistory):
+    HISTORY_VERSION = 1
+
+    def __init__(self, filename, max_commands=500):
+        super().__init__(filename)
+        self._max_commands = max_commands
+
+    def load_history_strings(self):
+        try:
+            with open(self.filename, 'r') as f:
+                commands = json.load(f).get('commands', [])
+            return reversed(commands)
+        except Exception as e:
+            LOG.debug(f'Exception on loading prompt history - {e}')
+            return []
+
+    def store_string(self, string):
+        history = {'version': self.HISTORY_VERSION, 'commands': []}
+        try:
+            if os.path.exists(self.filename):
+                with open(self.filename, 'r') as f:
+                    history = json.load(f)
+            history['commands'].append(string)
+            history['commands'] = history['commands'][-self._max_commands:]
+            with open(self.filename, 'w') as f:
+                json.dump(history, f)
+        except Exception as e:
+            LOG.debug(f'Exception on storing prompt history - {e}')
```

Comment:
Oh instead we should just use `exc_info` as it will log the Traceback to. So something like this: https://github.com/aws/aws-cli/blob/ab41dc3429137dd964670e9bc342590ca4b7bf2b/awscli/customizations/sso/logout.py#L92